app/uk/gov/hmrc/cataloguefrontend/metrics/model/ServiceProgressMetrics.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cataloguefrontend.metrics.model
+
+import play.api.libs.json.{Json, Reads}
+
+final case class ServiceProgressMetrics(
+  name: String,
+  group: String,
+  repository: String,
+  isHappy: Boolean
+)
+
+object ServiceProgressMetrics{
+  implicit val reads: Reads[ServiceProgressMetrics] = Json.reads[ServiceProgressMetrics]
```

Comment:
Rather than making these implicits in the companion object, since they are not facts about `ServiceProgressMetrics` but relate to certain endpoints, I would remove the `implicit` and bring them into scope where they apply. It also makes it easier to see how they are being used.

Reply:
Rather than using the macro which couples the internal model to the externally defined API, I think it's preferable to use explicit mapping.

Reply:
I'd be fully in agreement with you if this was an API exposing a resource to multiple clients and we wanted to evolve the presentation for each of them separately, say in a content supply chain or a financial API, but I think it's fine to treat codecs instances as coherent here, specially after us opting for having a separate module `metrics` for all this data and other modules not being likely to re-use the models.